### PR TITLE
Fix bug where sendRequest returns undefined when component unmounts

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/v3-beta/issues/82
https://github.com/exercism/v3-beta/issues/112

I just found out that when a component unmounts, our `sendRequest` function returns `undefined`. This is why tooltips don't load on hovering out of them quickly. Also, I think this is why the notification dropdown doesn't work, since we've unmounted the component when we change pages.

There are some things that might break from this, and these might at the top of my head:

1. Cancelling a test run.

I will handle this case at the same time as this issue: https://github.com/exercism/v3-beta/issues/123

2. Scratchpad 

When I was looking at the scratchpad code, I noticed that we weren't using `useRequestQuery`. I will make a separate issue for this.

### CODE REVIEW

File: app/javascript/components/mentoring/session/Scratchpad.tsx [hunk 1]
```diff
@@ -38,45 +34,48 @@ export const Scratchpad = ({
...
+        }
+      },
+    }
+  )
+
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault()
+
+      setError('')
```
Comment (iHiD) [inline app/javascript/components/mentoring/session/Scratchpad.tsx:65]: Thoughts on doing `setError(undefined)` instead of `''`?

**Comment by iHiD** [comment]
@kntsoriano Should we also remove `use-is-mounted` from the package.json?

**Comment by kntsoriano** [comment]
@iHiD We still use it in some cases.

**Comment by iHiD** [comment]
OK, great :)

**Pull request merged**